src/hooks/usePreferredCurrency.ts

```diff
@@ -0,0 +1,33 @@
+import {useMemo} from 'react';
+import {useOnyx} from 'react-native-onyx';
+import type {ValueOf} from 'type-fest';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+
+type PreferredCurrency = ValueOf<typeof CONST.PAYMENT_CARD_CURRENCY>;
+
+/**
+ * Get user's preferred currency in the following order:
+ *
+ * 1. Default card currency
+ * 2. User's local currency (if it's a valid payment card currency)
+ * 3. USD (default currency)
+ *
+ */
+function usePreferredCurrency(): PreferredCurrency {
+    const [personalDetails] = useOnyx(ONYXKEYS.PERSONAL_DETAILS_LIST);
+    const [session] = useOnyx(ONYXKEYS.SESSION);
+    const [fundList] = useOnyx(ONYXKEYS.FUND_LIST);
+
+    const defaultCardCurrency = useMemo(() => Object.values(fundList ?? {}).find((card) => card.isDefault)?.accountData?.currency, [fundList]);
```

Comment:
```suggestion
    const paymentCardCurrency = useMemo(() => Object.values(fundList ?? {}).find((card) => card.accountData?.additionalData?.isBillingCard)?.accountData?.currency, [fundList]);
```

Reply:
The defaultCard is for P2P wallet, we want to identify the payment card (backend data will match my suggestion after https://github.com/Expensify/App/issues/43136 is completed)

Reply:
Got it, so basically we have to wait for https://github.com/Expensify/App/issues/43136 to be closed so we can merge this?

Reply:
No I don't think we need to wait for it to finish/deploy, we can make the update here ahead of time and merge since we know what the shape _will_ be

Reply:
Updated!